### PR TITLE
Exit if uid=0, add --force flag to override

### LINKED ISSUE #3138: minikube should fail early if executed as root
This is a bad practice, and could address some awkward connectivity and authentication issues, like #2548

### PR DESCRIPTION
**Pull request by tstromberg**
New behavior:

```
sudo ./out/minikube start
😄  minikube v1.3.1 on Darwin 10.14.6
🛑  The virtualbox driver should not be used with root privileges.
💡  Tip: To remove this root owned cluster, run: sudo minikube delete
💣  Exiting
```

or:

```
sudo ./out/minikube start --force 
😄  minikube v1.3.1 on Darwin 10.14.6
⚠️  WARNING: The virtualbox driver should not be used with root privileges.
💿  Downloading VM boot image ...
```

Closes #3138
Obsoletes #5179

### CODE REVIEW

**Comment by tstromberg** [comment]
@minikube-bot OK to test

File: cmd/minikube/cmd/start.go [hunk 1]
```diff
@@ -502,25 +505,33 @@ func selectImageRepository(mirrorCountry string, k8sVersion string) (bool, strin
...
-		}
+	useForce := viper.GetBool(force)
 
-	} else {
-		glog.Errorf("Error getting the current user: %v", err)
+	if d == constants.DriverNone && u.Uid != "0" && !useForce {
+		exit.WithCodeT(exit.Permissions, `The "{{.driver_name}}" driver requires root privileges. Please run minikube using 'sudo minikube --vm-driver={{.driver_name}}'.`, out.V{"driver_name": d})
 	}
 
+	if d == constants.DriverNone || u.Uid != "0" {
```
Comment (afbjorklund) [inline cmd/minikube/cmd/start.go]: Wouldn't this always exit, for driver none ? Or maybe that is the "happy path" of validateUser

Comment (afbjorklund) [reply cmd/minikube/cmd/start.go]: Never mind, the return will just go the same place as before (the checks were reversed)

**Comment by afbjorklund** [comment]
This might lead to some support issues, since configuration is unreadable (owned by root)

**Comment by tstromberg** [comment]
> This might lead to some support issues, since configuration is unreadable (owned by root)

Good point. Let me add something to detect if there is an existing config and offer up a solution.

**Pull request merged**